src/init.cpp

```diff
@@ -237,9 +237,8 @@ void clvk_global_state::init_executors() {
 void clvk_global_state::term_executors() { delete m_thread_pool; }
 
 clvk_global_state::clvk_global_state() {
-    init_config_from_env_only();
-    init_logging();
     init_config();
+    init_logging();
```

Comment:
I understand that parsing the env twice is a workaround more than a clean implementation.

The goal is to be able to have logs when running `parse_config_file`. So we need the logging to be initialized before it. And to have any logs, we need to parse the env.
Then we need to parse it one more time to override things that could have been set by `parse_config_file`.

Reply:
Makes sense, thanks. I did bump into this when I first implemented config handling and solved it by not using logging at all during config parsing. I'll revert my changes and add a comment for now.